src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
Such much comments... read the first few

`DataValueDeserializer` was added to allow getting rid of these static construction methods. Apparently no deprecation tag was added to these specific classes. Doing so should not be an issue. Perhaps removing them altogether also is not? If migration to `DataValueDeserializer` has been completed, you can just update the code in `DataValueDeserializer` and ditch these methods. About time after 4(?) years.

I agree that generally mentioning a class from a component you do not even depend on is either weird or flat out bad. Since the responsibility for this serialization and deserilaization stuff was moved to another component, it does seem like the sensible thing to do here. And anyway, no need to spend lots of time on writing the best comment ever if you just get rid of it and the code it goes along with in the near future (which is what I hope and suggest you do).